sled-agent/types/src/early_networking.rs

```diff
@@ -0,0 +1,606 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Full-admission, I'm sorta skimming over this rather than reading line-by-line because it looks mostly copied

Reply:
Yeah, that's fine since it's almost entirely code movement, there are no functional changes so it's more just making sure the code organization looks fine. Thanks!